python/sglang/srt/managers/schedule_batch.py

```diff
@@ -247,8 +247,11 @@ def __init__(
         self.embedding = None
 
         # Constrained decoding
-        self.regex_fsm: RegexGuide = None
+        self.regex_fsm: Optional[RegexGuide] = None
         self.regex_fsm_state: int = 0
+        self.regex_bnf: Optional[GrammarMatcher] = None
+
+        self.allow_jump_forward = False
```

Comment:
turn on this?

Reply:
Thank you for your feedback!

We've carefully reviewed and implemented the suggested changes. Specifically, we've now built an abstraction of `Grammar` to handle constrained decoding more elegantly.